Source/diablo.cpp

```diff
@@ -1798,61 +1798,47 @@ void __fastcall PressChar(int vkey)
 
 void __cdecl LoadLvlGFX()
 {
-	unsigned char *v0; // eax
-	char *v1; // ecx
-	unsigned char *v2; // eax
-	char *v3; // ecx
-	unsigned char *v4; // eax
-	char *v5; // ecx
-
-	if ( !leveltype )
+	switch (leveltype)
 	{
+	case DTYPE_TOWN:
 		pDungeonCels = LoadFileInMem("Levels\\TownData\\Town.CEL", 0);
 		pMegaTiles = LoadFileInMem("Levels\\TownData\\Town.TIL", 0);
-		v4 = LoadFileInMem("Levels\\TownData\\Town.MIN", 0);
-		v5 = "Levels\\TownData\\TownS.CEL";
-		goto LABEL_14;
-	}
-	if ( leveltype == 1 )
-	{
+		*(_DWORD*)&dpiece_defs[0].blocks = (_DWORD)LoadFileInMem("Levels\\TownData\\Town.MIN", 0);

```

Comment:
This really irks me. `LoadFileInMem` returns a pointer to the actual file, but it's being cast to 2 shorts. I'm confident that `dpiece_defs` is incorrectly typed, should probably be a pointer to whatever that `.MIN` format is.

Reply:
Yes, it is incorrectly typed. @galaxyhaxz wanted to change this in the decompilation first, that's why I haven't changed it here.